### PR TITLE
bugfix: fix ApplicationContext already closed problem when Seata server using ShutdownHook to destroy.

### PR DESCRIPTION
**Pull request by Pinocchio2018**
- [x] I have registered the PR [changes](https://github.com/seata/seata/tree/develop/changes).

### Ⅰ. Describe what this PR did
in develop version，when ShutdownHook doing destroyAll，an Exception happened  because ApplicationContext is already closed
```java
14:56:22.666 ERROR --- [             ShutdownHook] i.s.core.rpc.netty.NettyServerBootstrap  : shutdown execute error:org.springframework.boot.web.servlet.context.AnnotationConfigServletWebServerApplicationContext@5a4c638d has been closed already
==>
java.lang.IllegalStateException: org.springframework.boot.web.servlet.context.AnnotationConfigServletWebServerApplicationContext@5a4c638d has been closed already
	at org.springframework.context.support.AbstractApplicationContext.assertBeanFactoryActive(AbstractApplicationContext.java:1093) ~[spring-context-5.2.14.RELEASE.jar:5.2.14.RELEASE]
	at org.springframework.context.support.AbstractApplicationContext.getBean(AbstractApplicationContext.java:1126) ~[spring-context-5.2.14.RELEASE.jar:5.2.14.RELEASE]
	at io.seata.spring.boot.autoconfigure.provider.SpringBootConfigurationProvider.get(SpringBootConfigurationProvider.java:100) ~[classes/:na]
	at io.seata.spring.boot.autoconfigure.provider.SpringBootConfigurationProvider.get(SpringBootConfigurationProvider.java:85) ~[classes/:na]
	at io.seata.spring.boot.autoconfigure.provider.SpringBootConfigurationProvider.access$200(SpringBootConfigurationProvider.java:46) ~[classes/:na]
	at io.seata.spring.boot.autoconfigure.provider.SpringBootConfigurationProvider$1.intercept(SpringBootConfigurationProvider.java:61) ~[classes/:na]
	at io.seata.config.FileConfiguration$$EnhancerByCGLIB$$862af1eb.getConfig(<generated>) ~[classes/:na]
	at io.seata.discovery.registry.nacos.NacosRegistryServiceImpl.getServiceName(NacosRegistryServiceImpl.java:214) ~[classes/:na]
	at io.seata.discovery.registry.nacos.NacosRegistryServiceImpl.unregister(NacosRegistryServiceImpl.java:92) ~[classes/:na]
	at io.seata.core.rpc.netty.NettyServerBootstrap.shutdown(NettyServerBootstrap.java:187) ~[classes/:na]
	at io.seata.core.rpc.netty.AbstractNettyRemotingServer.destroy(AbstractNettyRemotingServer.java:126) ~[classes/:na]
	at io.seata.core.rpc.ShutdownHook$DisposablePriorityWrapper.destroy(ShutdownHook.java:113) ~[classes/:na]
	at io.seata.core.rpc.ShutdownHook.destroyAll(ShutdownHook.java:82) ~[classes/:na]
	at io.seata.core.rpc.ShutdownHook.run(ShutdownHook.java:66) ~[classes/:na]
<==
```

### Ⅱ. Does this pull request fix one issue?

fix https://github.com/seata/seata/issues/4028

### Ⅲ. Why don't you add test cases (unit test/integration test)? 

I don't know how to write one in this problem.

### Ⅳ. Describe how to verify it

1. checkout develop branch in idea，
2. using nacos as registry center 
3. start `ServerApplication` in debug mode
4. stop the seata server, check if exceptions happens

### CODE REVIEW

File: server/src/main/java/io/seata/server/ServerRunner.java [hunk 1]
```diff
@@ -51,4 +60,15 @@ public void run(String... args) {
     public boolean started() {
         return started;
     }
+
+    @Override
+    public void destroy() throws Exception {
+        LOGGER.debug("destoryAll starting");
```
Comment (a364176773) [inline server/src/main/java/io/seata/server/ServerRunner.java:66]: 判断输出级别

File: server/src/main/java/io/seata/server/ServerRunner.java [hunk 2]
```diff
@@ -15,21 +15,30 @@
...
  */
 @Component
-public class ServerRunner implements CommandLineRunner {
+public class ServerRunner implements CommandLineRunner, DisposableBean {
 
     private static final Logger LOGGER = LoggerFactory.getLogger(ServerRunner.class);
 
     private boolean started = Boolean.FALSE;
 
+    private static final List<Disposable> disposableList = new CopyOnWriteArrayList<>();
```
Comment (a364176773) [inline server/src/main/java/io/seata/server/ServerRunner.java:37]: 大写

**Review by a364176773 — APPROVED** [review]
LGTM

**Review by slievrly — COMMENTED** [review]
LGTM

**Pull request merged**